src/api/app/controllers/webui/watched_items_controller.rb

```diff
@@ -26,12 +26,12 @@ def toggle
   private
 
   def set_item
-    @item = if params[:package]
-              Package.find_by_project_and_name(params[:project], params[:package])
-            elsif params[:project]
-              Project.find_by(name: params[:project])
-            elsif params[:number]
-              BsRequest.find_by(number: params[:number])
+    @item = if params[:project_name] && params[:package_name]
+              Package.find_by_project_and_name(params[:project_name], params[:package_name])
+            elsif params[:project_name]
+              Project.find_by(name: params[:project_name])
+            elsif params[:request_number]
```

Comment:
Watching a request doesn't work as this should be `params[:number]`:

```suggestion
            elsif params[:number]
```

This should have been caught by a spec. Nothing is checking if this controller works for requests.

Reply:
It's updated and I've added some new specs for this. :slightly_smiling_face: